promdens.py

```diff
@@ -495,29 +557,29 @@ def windowing(self):
-parser.add_argument("-tu", "--tdm_units", default='a.u.',
-                    help="Units in which magnitudes of transition dipole moments (|mu_ij|) are provided. Options: 'a.u.', 'debye'.")
-parser.add_argument("-w", "--omega", default=0.1, type=float, help="Frequency of the field in a.u.")
+                         "PDA for windowing (PDAW) to generate weights and convolution parameters.")
+parser.add_argument("-n", "--nsamples", default=0, type=positive_int,
+                    help="Number of data points from the input file considered. By default all initial conditions provided in the input file are taken.")
+parser.add_argument("-np", "--npsamples", default=1000, type=positive_int, help="Number of initial conditions generated with PDA.")
+parser.add_argument("-ns", "--nstates", default=1, type=positive_int, help="Number of excited states considered.")
+parser.add_argument("-ft", "--file_type", choices=FILE_TYPES, default='file', help="Input file type.")
+parser.add_argument("-p", "--plot", action="store_true", help="Plot the input data and calculated results and save them as png images.")
+parser.add_argument("-eu", "--energy_units", choices=ENERGY_UNITS, default='a.u.', help="Units in which excitation energies are provided.")
+parser.add_argument("-tu", "--tdm_units", choices=TDM_UNITS, default='a.u.',
+                    help="Units in which magnitudes of transition dipole moments (|mu_ij|) are provided.")
+parser.add_argument("-w", "--omega", required=True, type=positive_float, help="Frequency of the field in a.u.")
 parser.add_argument("-lch", "--linear_chirp", default=0.0, type=float, help="Linear chirp [w(t) = w+lch*t] of the field frequency in a.u.")
-parser.add_argument("-f", "--fwhm", default=10.0, type=float,
+parser.add_argument("-f", "--fwhm", required=True, type=positive_float,
                     help="Full Width at Half Maximum (FWHM) parameter for the pulse intensity envelope in fs.")
 parser.add_argument("-t0", "--t0", default=0.0, type=float, help="Time of the field maximum in fs.")
-parser.add_argument("-env", "--envelope_type", default='gauss', help="Field envelope type. Options: 'gauss', 'lorentz', 'sech', 'sin', 'sin2'.")
-parser.add_argument("-neg", "--neg_handling", default='error',
-                    help="Procedures how to handle negative probabilities. Options: 'error', 'ignore', 'abs'.")
-parser.add_argument("-s", "--seed", default=None, type=int,
-                    help="Seed for the random number generator. Default (None) generates random seed from OS.")
+parser.add_argument("-env", "--envelope_type", choices=ENVELOPE_TYPES, default='gauss', help="Field envelope type.")
+parser.add_argument("-neg", "--neg_handling", choices=NEG_PROB_HANDLING, default='error',
+                    help="Procedures how to handle negative probabilities.")
+parser.add_argument("-s", "--seed", default=None, type=positive_int,
```

Comment:
I checked that the seed must be positive, otherwise numpy throws an error. It does allow 0 (and we reject it here) but that seems like a minor thing.